pkg/sentry/fs/fsutil/cache_reclaim.go

```diff
@@ -0,0 +1,350 @@
+package fsutil
+
+import (
+        "sync"
+        "sync/atomic"
+        "time"
+        "runtime"
+
+        "gvisor.dev/gvisor/pkg/ilist"
+        "gvisor.dev/gvisor/pkg/log"
+        "gvisor.dev/gvisor/pkg/sentry/context"
+        "gvisor.dev/gvisor/pkg/sentry/memmap"
+        "gvisor.dev/gvisor/pkg/sentry/pgalloc"
+        "gvisor.dev/gvisor/pkg/sentry/platform"
+        "gvisor.dev/gvisor/pkg/sentry/usermem"
+)
+
+const (
+        LRU_ACTIVE_LIST = 0
+        LRU_INACTIVE_LIST = 1
+        LRU_LIST_NUM = 2
+)
+
+type LruEntry struct {
+	ilist.Entry
+	fr platform.FileRange
+	mr memmap.MappableRange
+	c *CachingInodeOperations
+	accessed bool
+	lru int
+}
+
+type LruManager struct {
+	destroyed bool
+	forceReclaim bool
+
+	// LRULists store the current file ranges consumed by all the files
+        // with the minimum granularity of 4K
+        // LRUMap records the mappings from the start of file range to its entry
+        //
+        // LRULists and LRUMap are protected by LRUMapMu
+        lists [LRU_LIST_NUM]ilist.List
+	length [LRU_LIST_NUM]int
+        mappings map[uint64]*LruEntry
+        lruMu sync.RWMutex
+
+	f *pgalloc.MemoryFile
+}
+
+func NewLruEntry(fr platform.FileRange, mr memmap.MappableRange, c *CachingInodeOperations) *LruEntry {
+        return &LruEntry{fr: fr, mr: mr, c: c, accessed: true, lru: LRU_ACTIVE_LIST}
+}
+
+func update(e *LruEntry, fr platform.FileRange, mr memmap.MappableRange, c *CachingInodeOperations) {
```

Comment:
Why not a function of LruEntry? I guess I don't really understand this function. The name "update" is strange: it's actually reassigning everything meaningful that is in the LruEntry. This definitely needs an explanation.

Reply:
Yes, it should be, and I will change it.

Sorry that I just find this method is not so correct as it will miss some ranges. I will recheck it and get a new version.

Reply:
After some examinations, I decide to change the cache tracking granularity to page size instead of changeable range, to avoid missing ranges. During read cache hit, we'll use dedicated `Access` function to touch the cache so as to avoid an extra level of locking in `Insert` path. The performance seems to be the same, but the size of metadata(the entries in lru list) increases.

I'll update the description in the PR soon, or should I put it in next PR?